src/main/java/uk/gov/justice/digital/nomis/service/XtagEventsService.java

```diff
@@ -61,6 +61,17 @@ private OffenderEvent addAdditionalEventData(final OffenderEvent oe) {
                     .orElse(null);
             oe.setOffenderIdDisplay(nomsId);
         }
+        else if ("EXTERNAL_MOVEMENT_RECORD-INSERTED".equals(oe.getEventType())){
+            offenderService.getExternalMovement(oe.getBookingId(), oe.getMovementSeq()).ifPresent(em -> {
+                final var offenderBooking = em.getId().getOffenderBooking();
+                oe.setOffenderIdDisplay(offenderBooking.getOffender().getOffenderIdDisplay());
```

Comment:
How many SQL are generated for this.?

Reply:
2 statements. Unfortunately, unless I get the nomsId the rest of it is no help form Pathfinders point of view